storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -884,26 +955,8 @@ private static void unTarUsingJava(File inFile, File untarDir,
                 entry = tis.getNextTarEntry();
             }
         } finally {
-            cleanup(tis, inputStream);
-        }
-    }
-
-    /**
-     * Close the Closeable objects and <b>ignore</b> any {@link IOException} or
-     * null pointers. Must only be used for cleanup in exception handlers.
-     *
-     * @param closeables the objects to close
-     */
-    private static void cleanup(java.io.Closeable... closeables) {
-        for (java.io.Closeable c : closeables) {
-            if (c != null) {
-                try {
-                    c.close();
-                } catch (IOException e) {
-                    LOG.debug("Exception in closing " + c, e);
-
-                }
-            }
+            tis.close();
+            inputStream.close();
```

Comment:
Do we need null checks here when we call close? I think something could go wrong before they are assigned.